streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -42,32 +42,38 @@ public StreamThreadStateStoreProvider(final StreamThread streamThread) {
 
     @SuppressWarnings("unchecked")
     @Override
-    public <T> List<T> stores(final String storeName, final QueryableStoreType<T> queryableStoreType) {
+    public <T> List<T> stores(final String storeName,
+                              final QueryableStoreType<T> queryableStoreType,
+                              final boolean includeStandbyAndRecovering) {
         if (streamThread.state() == StreamThread.State.DEAD) {
             return Collections.emptyList();
         }
-        if (!streamThread.isRunningAndNotRebalancing()) {
-            throw new InvalidStateStoreException("Cannot get state store " + storeName + " because the stream thread is " +
-                    streamThread.state() + ", not RUNNING");
-        }
-        final List<T> stores = new ArrayList<>();
-        for (final Task streamTask : streamThread.tasks().values()) {
-            final StateStore store = streamTask.getStore(storeName);
-            if (store != null && queryableStoreType.accepts(store)) {
-                if (!store.isOpen()) {
-                    throw new InvalidStateStoreException("Cannot get state store " + storeName + " for task " + streamTask +
-                            " because the store is not open. The state store may have migrated to another instances.");
-                }
-                if (store instanceof TimestampedKeyValueStore && queryableStoreType instanceof QueryableStoreTypes.KeyValueStoreType) {
-                    stores.add((T) new ReadOnlyKeyValueStoreFacade((TimestampedKeyValueStore<Object, Object>) store));
-                } else if (store instanceof TimestampedWindowStore && queryableStoreType instanceof QueryableStoreTypes.WindowStoreType) {
-                    stores.add((T) new ReadOnlyWindowStoreFacade((TimestampedWindowStore<Object, Object>) store));
-                } else {
-                    stores.add((T) store);
+        final StreamThread.State state = streamThread.state();
+        if (includeStandbyAndRecovering ? state.isRunning() : state == StreamThread.State.RUNNING) {
```

Comment:
this is creative :)

Reply:
state.isRunning() is as below

```
public boolean isRunning() {
            return equals(RUNNING) || equals(STARTING) || equals(PARTITIONS_REVOKED) || equals(PARTITIONS_ASSIGNED);
        }
```

if its not too much work, we can rename to something like `state.isAlive()` , that captures what we want to check .. your call.

Reply:
In the case where `includeStandbyAndRecovering=false` and the state is `RUNNING`.. We still want to disallow queries on standbys? from code below, it seems like we will enter the if block and loop over all active and standby tasks?

Reply:
really nice catch! Thanks.

Regarding isRunning, I had a similar thought, but couldn't remember if StreamThread.State is exposed in the public API. After some code analysis, I don't think it is, so I've renamed it.